src/gluonts/transform/field.py

```diff
@@ -39,23 +39,31 @@ def __init__(self, mapping: Dict[str, str]) -> None:
 
     def transform(self, data: DataEntry):
         for key, new_key in self.mapping.items():
-            if key not in data:
-                continue
-            assert new_key not in data
-            data[new_key] = data[key]
-            del data[key]
+            if key in data:
+                # no implicit overriding
+                assert new_key not in data
+                data[new_key] = data[key]
+                del data[key]
         return data
 
 
 class RemoveFields(SimpleTransformation):
+    """"
+    Remove field names if present.
+
+    Parameters
+    ----------
+    field_names
+        List of names of the fields that will be removed
+    """
+
     @validated()
     def __init__(self, field_names: List[str]) -> None:
         self.field_names = field_names
 
     def transform(self, data: DataEntry) -> DataEntry:
         for k in self.field_names:
-            if k in data.keys():
-                del data[k]
+            data.pop(k, None)
```

Comment:
This change is in contrast with the docstring I think ("if present"): if the field is absent, `pop` will raise a `KeyError`

Reply:
If it's not present it will return `None` and should not fail. We don't use the return of the `pop` method, so the implementation should be equivalent?